Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/SparkServerlessClusterProvisionCtrlProvider.java

```diff
@@ -0,0 +1,161 @@
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.azure.sparkserverless.serverexplore;
+
+import com.microsoft.azure.hdinsight.common.mvc.IdeSchedulers;
+import com.microsoft.azure.hdinsight.common.mvc.SettableControl;
+import com.microsoft.azure.hdinsight.sdk.cluster.ProvisionableCluster;
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessAccount;
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessCluster;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+import com.microsoft.azuretools.azurecommons.helpers.StringHelper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import rx.Observable;
+
+public class SparkServerlessClusterProvisionCtrlProvider {
+
+    @NotNull
+    private SettableControl<SparkServerlessClusterProvisionSettingsModel> controllableView;
+
+    @NotNull
+    private IdeSchedulers ideSchedulers;
+
+    @NotNull
+    private AzureSparkServerlessAccount account;
+    public SparkServerlessClusterProvisionCtrlProvider(
+            @NotNull SettableControl<SparkServerlessClusterProvisionSettingsModel> controllableView,
+            @NotNull IdeSchedulers ideSchedulers,
+            @NotNull AzureSparkServerlessAccount account) {
+        this.controllableView = controllableView;
+        this.ideSchedulers = ideSchedulers;
+        this.account = account;
+    }
+
+    @NotNull
+    public int getCalculatedAU(@Nullable String masterCoresStr,
+                               @Nullable String workerCoresStr) {
+        int masterCores = NumberUtils.toInt(masterCoresStr);
+        int workerCores = NumberUtils.toInt(workerCoresStr);
+        return masterCores + workerCores;
+    }
+
+    // TODO: Update getTotalAU
+    @NotNull
+    public int getTotalAU(@Nullable String adlAccount) {
+        // TODO: adlAccount validity check
+        return 0;
+    }
+
+    // TODO: Update getAvailableAU
+    @NotNull
+    public int getAvailableAU(@Nullable String adlAccount) {
+        // TODO: adlAccount validity check
+        return 0;
+    }
+
+    public Observable<SparkServerlessClusterProvisionSettingsModel> validateAndProvision() {
+        return Observable.just(new SparkServerlessClusterProvisionSettingsModel())
+                .doOnNext(controllableView::getData)
+                .observeOn(ideSchedulers.processBarVisibleAsync("Validating the cluster settings..."))
+                .map(toUpdate -> {
+                    String clusterName = toUpdate.getClusterName();
```

Comment:
Consider to split the anonymous function into small ones.

Reply:
Will fix in the next version

---
In reply to: [191116128](https://github.com/Microsoft/azure-tools-for-java/pull/1665#discussion_r191116128) [](ancestors = 191116128)